tests/php/StoreApi/Routes/CartExtensions.php

```diff
@@ -6,16 +6,50 @@
 namespace Automattic\WooCommerce\Blocks\Tests\StoreApi\Routes;
 
 use Automattic\WooCommerce\Blocks\Tests\StoreApi\Routes\ControllerTestCase;
+use Automattic\WooCommerce\Blocks\Tests\Helpers\FixtureData;
 
 /**
  * Cart Controller Tests.
  */
 class CartExtensions extends ControllerTestCase {
 
+	/**
+	 * Setup test products data. Called before every test.
+	 */
+	public function setUp() {
+		parent::setUp();
+
+		$fixtures = new FixtureData();
+
+		$this->product = $fixtures->get_simple_product(
+			array(
+				'name'          => 'Test Product 1',
+				'regular_price' => 10,
+			)
+		);
+
+		wc_empty_cart();
+
+		wc()->cart->add_to_cart( $this->product->get_id(), 1 );
+
+		woocommerce_store_api_register_update_callback(
+			array(
+				'namespace' => 'valid-test-plugin',
+				'callback'  => function() {
+					add_action(
+						'woocommerce_cart_calculate_fees',
+						function() {
+							wc()->cart->add_fee( 'Surcharge', 10, true, 'standard' );
+						}
+					);
```

Comment:
We're using fees here because they're the only core cart method that doesn't trigger its own recalc, however, they only work if you add them within this action.